### PR TITLE
Check GitHub Issue Count

### LINKED ISSUE #75: Real-time stats for outstanding issues
Because it's really hard to tell how many issues are actually outstanding without clicking through.
- likely requires a backend - RIP static site
- what do we do when a site has no outstanding issues?

### PR DESCRIPTION
**Pull request by dahlbyk**
Nerd-sniped myself tonight: http://dahlbyk-demo.github.io/up-for-grabs.net/

This fixes #75 with a client-side GitHub API request. Rate limited to 60/hr, at which point you'll see something like this:
github-rate-limit

Commits are a disaster - will clean up - but...thoughts on the general approach?

A follow-up up-for-grabs feature would be to cache issue counts in local storage with some expiration policy (an hour? when rate limit resets?).

### CODE REVIEW

**Comment by shiftkey** [comment]
:metal: I like the approach. 

> A follow-up up-for-grabs feature would be to cache issue counts in local storage with some expiration policy (an hour? when rate limit resets?).

I'd suggest a couple of days - I'd rather us err on the side of keeping stale data around a bit instead of abusing the API limits unnecessarily..

File: index.html [hunk 1]
```diff
@@ -112,7 +112,7 @@ <h2 id="project_tagline">explore open source projects and <a href="http://nikcod
     <!-- FOOTER  -->
     <div id="footer_wrap" class="outer">
       <footer class="inner">
-        <p class="copyright">Up For Grabs maintained by <a href="https://github.com/dahlbyk">dahlbyk</a></p>
+        <p class="copyright">Up For Grabs maintained by <a href="https://github.com/dahlbyk">dahlbyk</a>, <a href="https://github.com/shiftkey">shiftkey</a> &amp; <a href="https://github.com/jrusbatch">jrusbatch</a></p>
```
Comment (shiftkey) [inline index.html:115]: :heart_eyes:

File: javascripts/main.js [hunk 2]
```diff
@@ -35,6 +35,28 @@
   });
 
   $(function() {
+    $(document).on("mouseenter focus mouseleave", '.label a', function(){
```
Comment (shiftkey) [inline javascripts/main.js]: This was hard to discover, could this be just the entire cell - aka `.projects tbody`?

**Comment by dahlbyk** [comment]
> I'd suggest a couple of days - I'd rather us err on the side of keeping stale data around a bit instead of abusing the API limits unnecessarily..

Or we may be able to just cache the ETag, since 304's don't seem to count against the rate limit:

File: javascripts/main.js (hunk 2 above)
Comment (dahlbyk) [inline javascripts/main.js]: Done.

File: stylesheets/stylesheet.css [hunk 3]
```diff
@@ -477,6 +477,33 @@ Custom Styles
...
 
+.projects .label .count {
+  background: none white;
+  border: 1px solid white;
+  border-radius: 8px;
+  color: #0090ff;
+  display: inline-block;
+  height: 16px;
+  line-height: 16px;
+  margin: -2px -4px 0 4px;
```
Comment (shiftkey) [inline stylesheets/stylesheet.css:488]: Not sure what the `-2px` here is for, but this looks weird in Chrome

screen shot 2015-02-09 at 8 42 32 pm

Putting this back to `0px` feels better:

screen shot 2015-02-09 at 8 42 06 pm

The text also seems to be slightly above the vertical center, but I Am Not A CSS Guru...

**Comment by shiftkey** [comment]
> Or we may be able to just cache the ETag, since 304's don't seem to count against the rate limit:

:thumbsup:

**Comment by shiftkey** [comment]
screen shot 2015-02-09 at 8 50 24 pm

For projects like NuGet which aren't on GitHub, should we show anything?

File: stylesheets/stylesheet.css (hunk 3 above)
Comment (dahlbyk) [inline stylesheets/stylesheet.css:488]: Hrm, this is what I see in Chrome (Windows):


And Firefox:

And IE:


All that said, I'm no CSS Guru either - styling was set for the spinner image first, then adjusted for the count. Probably better to go the other direction.

**Comment by dahlbyk** [comment]
> For projects like NuGet which aren't on GitHub, should we show anything?

Question mark with `title="Issue count not available."`?

**Comment by shiftkey** [comment]
> Question mark 

Yes!

File: stylesheets/stylesheet.css (hunk 3 above)
Comment (shiftkey) [inline stylesheets/stylesheet.css:488]: Oops, should have mentioned I was testing this in OS X.

:thumbsup: I'll chalk this one up to cross-browser hilarity and leave it for someone else to get sufficiently caremad about.

File: javascripts/main.js [hunk 4]
```diff
@@ -34,7 +34,56 @@
...
+  var issueCount = function(project) {
+    var a = $(project).find('.label a')
+      , gh = a.attr('href').match(/github.com(\/[^\/]+\/[^\/]+\/)labels\/([^\/]+)$/)
+      , url = gh && ('https://api.github.com/repos' + gh[1] + 'issues?labels=' + gh[2])
+      , count = a.find('.count');
+
+    if (!!count.length) { return; }
+
+    if (!gh) {
+      count = $('<span class="count" title="Issue count is only available for projects on GitHub.">?</span>').appendTo(a);
```
Comment (dahlbyk) [inline javascripts/main.js:73]: This is new

**Comment by dahlbyk** [comment]
Alrighty, updated to fetch issue counts when a project's `tbody` scrolls into view. Not optimized, not cached, but functional?

**Comment by Steven-Pingel** [comment]
Not sure if you are open to it but I tried my hand at using local storage to reduce some of the API calls.

PR: https://github.com/dahlbyk-demo/up-for-grabs.net/pull/1
DEMO: http://steven-pingel.github.io/up-for-grabs.net/#/

**Comment by dahlbyk** [comment]
> Add .gitattributes
> Add .editorconfig

Calling these out for feedback as well. Seems we've already committed LF with 2-space indent everywhere, so running with that.

**Comment by shiftkey** [comment]
> Calling these out for feedback as well. 

I ENDORSE THESE CONFIG FILES

**Comment by jkodroff** [comment]
I'd love to see this in production after clicking on a few projects with no issues.  Do you folks need any help getting this done?

**Comment by dahlbyk** [comment]
I'm not sure I ever asked for a final review, but...yeah, any feedback before this :ship:s?

**Comment by shiftkey** [comment]
@dahlbyk there's a few `fixup!` commit messages - are those supposed to be squashed away?

**Comment by shiftkey** [comment]
Aside from that, I'm :thumbsup: on the change - awesome work @dahlbyk!

**Comment by dahlbyk** [comment]
:8ball:

**Comment by adamralph** [comment]
:ok_hand:

**Pull request merged**